pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -411,7 +414,14 @@ func (lb *ListenerBuilder) buildHTTPConnectionManager(httpOpts *httpListenerOpts
 		filters = extension.PopAppend(filters, wasm, extensions.PluginPhase_AUTHN)
 		filters = append(filters, lb.authnBuilder.BuildHTTP(httpOpts.class)...)
 		filters = extension.PopAppend(filters, wasm, extensions.PluginPhase_AUTHZ)
-		filters = append(filters, lb.authzBuilder.BuildHTTP(httpOpts.class)...)
+
+		var httpauthz []*hcm.HttpFilter
+		if lb.node.Type == model.Waypoint {
+			httpauthz = lb.authzBuilder.BuildHTTPWithListener(httpOpts.class, httpOpts.routeConfig.Name)
```

Comment:
`routeConfig.Name` as "Listener" param seems unexpected. Maybe correct, though

Reply:
That was the param I found that had the listener name that also matched the dynamic_listeners[].name attribute from the config dump.